extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java

```diff
@@ -377,17 +379,25 @@ public void handle(RoutingContext event) {
                 if (!event.request().isEnded()) {
                     event.request().pause();
                 }
+
                 //we handle auth failure directly
                 event.remove(QuarkusHttpUser.AUTH_FAILURE_HANDLER);
+
                 VertxHttpExchange exchange = new VertxHttpExchange(event.request(), allocator, executorService, event,
                         event.getBody());
                 exchange.setPushHandler(VertxHttpRecorder.getRootHandler());
+
                 Optional<MemorySize> maxBodySize = httpConfiguration.limits.maxBodySize;
                 if (maxBodySize.isPresent()) {
                     exchange.setMaxEntitySize(maxBodySize.get().asLongValue());
                 }
                 Duration readTimeout = httpConfiguration.readTimeout;
                 exchange.setReadTimeout(readTimeout.toMillis());
+
+                UndertowOptionMap.Builder undertowOptions = UndertowOptionMap.builder();
+                undertowOptions.set(UndertowOptions.MAX_PARAMETERS, servletRuntimeConfig.maxParameters.orElse(0));
+                exchange.setUndertowOptions(undertowOptions.getMap());
```

Comment:
This map should be built once, not per-request.

Reply:
Ah, right.  Done.